src/controllers/news.ts

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable @typescript-eslint/type-annotation-spacing */
+/* eslint-disable eol-last */
+/* eslint-disable prettier/prettier */
+import News from '../models/news';
+import User from '../models/user';
+import {Request, Response, NextFunction} from 'express';
+import {createError} from '../utils/helpers';
+
+
+
+
+export const getNews = async (req: Request, res: Response) => {
+  try {
+    // console.log(req.query);
+    const queryObject =req.query;
+
+    const queryKeys= Object.keys(queryObject)
+    if(queryKeys.length ===0){
+      const news = await News.find({}).sort({publDate: 'desc'});
+      res.send(news)
+    }
+    if(queryObject.trending === 'true'){
+      const trendNews =await News.find({}).sort({clicks:'desc'})
+      res.send(trendNews)
+    }
+    else{
+      throw new Error('Query Not Matched')
+    }
+
+
+
+  } catch (e) {
+    res.status(500).send(e);
```

Comment:
You don't need to send the response here, just use `return next(e)` to the error to error handling middleware. You can check `user.ts` for reference. Same goes for all the functions.